src/core/core.controller.js

```diff
@@ -451,6 +460,10 @@ class Chart {
     config.update();
     me._options = config.createResolver(config.chartOptionScopes(), me.getContext());
 
+    if (me.aspectRatio !== me._aspectRatio) {
```

Comment:
Perhaps this should only happen if `maintainAspectRatio === false`, but not certain on that